dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -396,6 +512,10 @@ void WindowTransformAction::advanceFrameEnd()
         break;
     }
     case WindowFrame::BoundaryType::Offset:
+    {
+        std::tie(frame_end, frame_ended) = stepBackward(current_row, window_description.frame.end_offset);
```

Comment:
Maybe
```
if (partition_end)
    frame_ended = stepBackward(current_row, window_description.frame.end_offset);
```
is better?
Same as `stepForward`.

Reply:
> 

No bool variable is returned because the partition_start is always true for `stepForward`, so I think it's better to return the bool variable in `stepBackward`.